packages/11ty/_plugins/figures/tests/index.js

```diff
@@ -0,0 +1,51 @@
+require('module-alias/register')
+const { mock, test } = require('node:test')
+const assert = require('assert/strict')
+const Figure = require('../figure')
+const figureFixtures = require('./__fixtures__/figures/index.js')
+const iiifConfig = require('./__fixtures__/iiif-config.json')
+const Manifest = require('../iiif/manifest')
+
+const createManifestFromFigureFixture = async (figureFixtureName) => {
+  const {
+    dimensions,
+    figure: figureFixture,
+    manifest: manifestFixture
+  } = figureFixtures[figureFixtureName]
+  const { height, width } = dimensions
+  const figure = new Figure(iiifConfig, null, figureFixture)
+  figure.canvasHeight = height
+  figure.canvasWidth = width
+  const manifest = new Manifest(figure)
+  const manifestJson = await manifest.toJSON()
+  return {
+    manifestFixture,
+    manifestJson
+  }
+}
```

Comment:
@anderspollack this is an interesting approach for the unit testing the input/output; it seems to isolate that the for the given `figure.json` **input** data the `Figure` class will return an instance from which the `Manifest` will **output** the correct `manifest.json`. 

The integration test likely need to **mock** the file system, either using the [`mock-fs`](https://www.npmjs.com/package/mock-fs) package and for integration tests with a starter template using the [`unionfs`](https://github.com/streamich/unionfs) and [`memfs`](https://github.com/streamich/memfs) packages to merge the actual file system with the in-memory mocks.